easybuild/easyblocks/f/flang.py

```diff
@@ -0,0 +1,185 @@
+        CXX = os.path.join(self.llvm_obj_dir, 'bin', 'clang++')
+        FC = os.path.join(self.llvm_obj_dir, 'bin', 'flang')
+        LLVM_CONFIG = os.path.join(self.llvm_obj_dir, 'bin', 'llvm-config')
+        shlib_ext = get_shared_lib_ext()
+        LIBOMP = os.path.join(self.llvm_obj_dir, 'lib', 'libomp.%s' % shlib_ext)
+
+        options = "-DCMAKE_INSTALL_PREFIX=%s " % self.installdir
+        options += "-DCMAKE_C_COMPILER='%s' " % CC
+        options += "-DCMAKE_CXX_COMPILER='%s' " % CXX
+        options += "-DCMAKE_Fortran_COMPILER='%s' " % FC
+        options += "-DLLVM_CONFIG='%s' " % LLVM_CONFIG
+        options += "-DFLANG_LIBOMP='%s' " % LIBOMP
+        for option in additional_options:
+            options += "%s " % option
+        options += self.cfg['configopts']
+
+        self.log.info("Configuring")
+        run_cmd("cmake %s %s" % (options, src_dir), log_all=True)
+
+        self.log.info("Building")
+        if parallel:
+            run_cmd("make %s" % self.make_parallel_opts, log_all=True)
+        else:
+            run_cmd("make", log_all=True)
+
+    def build_step(self):
+        # First build llvm and the driver
+        super(EB_Flang, self).build_step()
+        if self.cfg['bootstrap']:
+            self.llvm_obj_dir = self.llvm_obj_dir_stage3
+        else:
+            self.llvm_obj_dir = self.llvm_obj_dir_stage1
+
+        # Build libpgmath with the temporary llvm
+        self.pgmath_build_dir = os.path.join(self.builddir, 'pgmath_obj')
+        self.build_with_temporary_llvm(
+            self.pgmath_build_dir,
+            os.path.join(self.flang_source_dir, 'runtime', 'libpgmath')
+            )
+        # Build flang with the temporary llvm
+        self.flang_build_dir = os.path.join(self.builddir, 'flang_obj')
+        shlib_ext = get_shared_lib_ext()
+        self.build_with_temporary_llvm(
+            self.flang_build_dir,
+            self.flang_source_dir,
+            parallel=False,  # Can misbehave in parallel
+            additional_options=[
+                '-DLIBPGMATH=%s' % os.path.join(self.pgmath_build_dir, 'lib', 'libpgmath.%s' % shlib_ext),
+                # Supposed to resolve a problem when linking with flang
+                '-DCMAKE_BUILD_WITH_INSTALL_RPATH=1',
+            ]
+        )
+
+    def install_step(self):
+        """Install flang binaries and necessary libraries."""
+        shlib_ext = get_shared_lib_ext()
+        change_dir(self.pgmath_build_dir)
+        super(EB_Clang, self).install_step()
+        change_dir(self.flang_build_dir)
+        super(EB_Clang, self).install_step()
+        # Copy over the flang executable from LLVM
+        copy_file(
+            os.path.join(self.llvm_obj_dir, 'bin', 'flang'),
+            os.path.join(self.installdir, 'bin', 'flang'),
+        )
+        # Also copy over the libomp libraries
+        copy_file(
+            os.path.join(self.llvm_obj_dir, 'lib', 'libomp.%s' % shlib_ext),
```

Comment:
Bit dubious about whether this is enough there are a number of libomp libraries in that directory but this is the only one that appears in the *Flang Binary Release File List* at https://github.com/flang-compiler/flang/releases

Reply:
I'm also concerned that an identical library (same source code) is created by Clang, are these compatible?